### DISCUSSION TITLE
HFSS Set Solution Type Options

### DISCUSSION THREAD

**Discussion by rjwilliams84** [ansys/pyaedt#2927] [Q&A] [answered]
Probably a simple solution but I can't find the answer in the documentation or examples.

I'm trying to set up HFSS and define the solution type as 'HFSS with Hybrid and Arrays', 'Modal' solution, with the 'Composite Excitation' option. All I can seem to do is change either 'Modal' or 'Terminal' using the following:

hfss=pyaedt.Hfss(projectname=project_name, designname=design_name ,solution_type="Modal")

where I've tried changing "Modal" after solution_type to other strings, such as "Terminal" (this works fine), "HFSS with Hybrid and Arrays" (does not do anything), and "Composite Excitation" (also does not do anything).

This leads me to believe that I'm not using the correct method to set these solution types/options, but as I mentioned in the beginning of this post I cannot find how to do this in the documentation or in any of the provided examples. I'm hoping someone can point me to what I need to do here.

Thanks.

**Comment by rjwilliams84** [comment]
If it helps, I'm looking for the pyaedt equivalent of:

oDesign.SetSolutionType("HFSS Hybrid Modal Composite", 
	[
		"NAME:Options",
		"EnableAutoOpen:="	, False
	])

but I can't find anything in the documentation

**Comment by maxcapodi78** [comment, marked as answer]
@rjwilliams84  you need the following code:
```
hfss=pyaedt.Hfss(projectname=project_name, designname=design_name ,solution_type="Modal")
hfss.hybrid = True
hfss.composite = True
```
Maybe we need to better document it. If you didn't find maybe is not clearly explained. @MaxJPRey  Maybe can you take care of it?

**Reply by rjwilliams84** [reply]
Thanks, Massimo.

I did find the hfss.composite command from the documentation, but did not understand how to use it. The documentation on that command is a bit sparse. I did not find the hfss.hybrid command when I was searching previously but after a quick check just now I see that command.

I have just verified that the code snippet you provided works on my machine for the desired effect. I think what it came down to for me is that for these command in particular, the documentation didn't have enough information for me to figure it out on my own.

**Reply by MaxJPRey** [reply]
@maxcapodi78 No problem. I will take care of it.
Indeed we need to clarify how to use `hybrid` and `composite` for `Modal` solution.
I am creating an issue to make sure that I do not forget about it.
